test/Microsoft.Sbom.Api.Tests/Executors/FileInfoWriterTests.cs

```diff
@@ -0,0 +1,223 @@
+
+namespace Microsoft.Sbom.Api.Tests.Executors;
+
+[TestClass]
+public class FileInfoWriterTests
+{
+    private Mock<ILogger> loggerMock;
+    private Mock<ISbomConfig> sbomConfigMock;
+    private Mock<ISbomPackageDetailsRecorder> sbomPackageDetailsRecorderMock;
+    private Mock<IManifestToolJsonSerializer> manifestToolJsonSerializerMock;
+    private FileInfoWriter testSubject;
+    private ManifestGeneratorProvider manifestGeneratorProvider;
+
+    [TestInitialize]
+    public void BeforeEach()
+    {
+        loggerMock = new Mock<ILogger>();
+        sbomPackageDetailsRecorderMock = new Mock<ISbomPackageDetailsRecorder>();
+        manifestToolJsonSerializerMock = new Mock<IManifestToolJsonSerializer>();
+
+        manifestGeneratorProvider = new ManifestGeneratorProvider(new IManifestGenerator[] { new TestManifestGenerator() });
+        manifestGeneratorProvider.Init();
+
+        testSubject = new FileInfoWriter(manifestGeneratorProvider, loggerMock.Object);
+
+        sbomConfigMock = new Mock<ISbomConfig>();
+        sbomConfigMock.SetupGet(x => x.ManifestInfo).Returns(Constants.TestManifestInfo);
+        sbomConfigMock.SetupGet(x => x.Recorder).Returns(sbomPackageDetailsRecorderMock.Object);
+        sbomConfigMock.SetupGet(x => x.JsonSerializer).Returns(manifestToolJsonSerializerMock.Object);
+    }
+
+    [TestCleanup]
+    public void AfterEach()
+    {
+        // Just verify nothing throws
+    }
+
+    [TestMethod]
+    public async Task Write_FileWithinDropPath_WritesToFilesSection()
+    {
+        // Arrange
+        var sbomConfigs = new[] { sbomConfigMock.Object };
+        var fileInfo = new InternalSbomFileInfo
+        {
+            Path = "test_file.txt",
+            IsOutsideDropPath = false,
+            FileTypes = new[] { FileType.SPDX },
+            Checksum = new[] { new Checksum { Algorithm = AlgorithmName.SHA256, ChecksumValue = "abc123" } }
+        };
+
+        var fileInfoChannel = Channel.CreateUnbounded<InternalSbomFileInfo>();
+        await fileInfoChannel.Writer.WriteAsync(fileInfo);
+        fileInfoChannel.Writer.Complete();
+
+        // Setup expectations
+        sbomPackageDetailsRecorderMock.Setup(m => m.RecordFileId(It.IsAny<string>()));
+
+        // Act
+        var (result, errors) = testSubject.Write(fileInfoChannel.Reader, sbomConfigs);
+
+        // Assert
+        var resultList = new List<JsonDocWithSerializer>();
+        await foreach (var item in result.ReadAllAsync())
+        {
+            resultList.Add(item);
+        }
+
+        var errorList = new List<FileValidationResult>();
+        await foreach (var error in errors.ReadAllAsync())
+        {
+            errorList.Add(error);
+        }
+
+        // Verify file was written to files section
+        Assert.AreEqual(1, resultList.Count);
+        Assert.AreEqual(0, errorList.Count);
+    }
+
+    [TestMethod]
+    public async Task Write_FileOutsideDropPath_DoesNotWriteToFilesSection()
```

Comment:
what's the difference between Write_FileOutsideDropPath_DoesNotWriteToFilesSection and Write_SpdxFileOutsideDropPath_RecordsSpdxFileId? If there's no difference, remove one of them